lib/validators/restrict-children-validator.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Validates restrictChildren properties option
+ */
+
+import { Validator } from './validator';
+import { DirectiveType, ParsedComponentsDefinition } from '../models';
+
+const PROPERTY = 'restrictChildren';
+const ADDITIONAL_PROPERTY = 'withContent';
+
+export class RestrictChildrenValidator implements Validator {
+
+    constructor(
+        private definition: ParsedComponentsDefinition,
+    ) {
+    }
+
+    private hasSlideshowDirective(parsedComponent: ParsedComponentsDefinition['components']['name']) : boolean {
+        return Object.values(parsedComponent.directives)
+        .some(directive => directive.type === DirectiveType.slideshow);
+    }
+
+    validate(
+        errorReporter: (errorMessage: string) => void,
+    ): boolean {
+        let valid = true;
+
+        Object.values(this.definition.components).forEach((parsedComponent: ParsedComponentsDefinition['components']['name']) => {
+            const isPresent = PROPERTY in parsedComponent.component && parsedComponent.component[PROPERTY];
+            if (!isPresent) {
+                if (this.hasSlideshowDirective(parsedComponent)) {
+                    errorReporter(`Component property "${PROPERTY}" must be defined in component "${parsedComponent.component.name}" because the ` + 
+                        `component contains slideshow directive`);
+                    valid = false;
+                }
+                return;
+            }
+            // check if all keys point to corrent component
+            // TODO if || {} is removed then ES2017 shows an error that parsedComponent.component[PROPERTY] may be undefined, why?
+            const propertyValue = parsedComponent.component[PROPERTY] || {};
```

Comment:
What if restrictChildren object is left empty? It would still pass.

Also, when a doc-slideshow directive is present, I would limit it to exactly one entry (as multiple entries don't really make sense).

Reply:
Updated the schema